### PR TITLE
server: actually scrub response

### PR DESCRIPTION
**Pull request by miekg**
Did all the worked, hooked it up wrongly :(

This also needs test, but those are hard(er) because we only receive
packets after they have been decoded; i.e. we never see the wirefmt.

Signed-off-by: Miek Gieben <miek@miek.nl>

### CODE REVIEW

**Comment by miekg** [comment]
oh nice. Our CI integration is gone after github's database failover...

**Comment by miekg** [comment]
Ok, tested added. PTAL.

File: request/request.go [hunk 1]
```diff
@@ -226,19 +226,15 @@ func (r *Request) SizeAndDo(m *dns.Msg) bool {
 	return true
 }
 
-// Scrub is a noop function, added for backwards compatibility reasons. The original Scrub is now called
-// automatically by the server on writing the reply. See ScrubWriter.
```
Comment (johnbelamaric) [inline request/request.go:230]: Was this comment incorrect? I mean, was the real Scrub never getting called in this case? So, we're not calling it twice now are we?

**Review by johnbelamaric — APPROVED** [review]
I think this is fine but one question...

**Comment by miekg** [comment]
Good point. It's the old comment I copied. I'll update

File: request/request.go (hunk 1 above)
Comment (miekg) [reply request/request.go:230]: Let me properly reply: no we're not calling it twice: Scrub is the only method left. The *old* Scrub was a noop, the new one is not.

Comment (johnbelamaric) [reply request/request.go:230]: Ok, so we weren't calling Scrub at all in the most recent release? Or just in some cases? That's what's confusing me. Maybe I just need to dig into the previous PR to understand it.

Comment (miekg) [reply request/request.go:230]: yes, we we're calling the noop function.

**Comment by johnbelamaric** [comment]
Eww. I am surprised more problems haven't come up.

/lgtm

**Comment by johnbelamaric** [comment]
Looks like this is breaking the rewrite tests.

**Comment by miekg** [comment]
huh, I don't understand that failure. The package send to Scrub also only contains the mx record @greenpau

**Comment by miekg** [comment]
broke *erratic* by adding any RRs if e.large wasn't set.

**Comment by greenpau** [comment]
@miekg, this is merged. Do you still need me to look over the test?

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "Re: [coredns/coredns] server: actua..." ]
>@miekg, this is merged. Do you still need me to look over the test?

thanks, but not needed. I made a booboo in the eratic plugin, luckily this test 
caught that :) All good know.

**Pull request merged**